tests/func/test_status.py

```diff
@@ -93,3 +93,17 @@ def test_status_on_pipeline_stages(tmp_dir, dvc, run_copy):
             "changed command",
         ],
     }
+
+
+def test_status_recursive(tmp_dir, dvc):
+    tmp_dir.gen({"dir": {"file": "text1", "subdir": {"file2": "text2"}}})
+    stages = dvc.add("dir", recursive=True, no_commit=True)
+
+    assert len(stages) == 2
+
+    assert dvc.status(targets=["dir"], recursive=True) == {
+        "dir/file.dvc": [{"changed outs": {"dir/file": "not in cache"}}],
```

Comment:
Tests are failing on windows because of the slashes. Please use `os.path.join` instead, that should fix it.